pyccel/parser/semantic.py

```diff
@@ -3452,6 +3594,13 @@ def _insert_obj(location, target, obj):
             else:
                 targets = expr.target
 
+            if import_init:
+                result  = FunctionCall(import_init,[],[])
+                targets.add(import_init.name)
```

Comment:
shouldn't the import_init be added to the `init_func` imports and not to the  module imports, to avoid name clach

Reply:
Hmm I hadn't considered name clashes here. I agree that `import_init` should perhaps be added to `init_func` if it is imported at the module level, but if it is imported inside a function then that is not equivalent. Maybe we could rename it if there is a collision?

Reply:
yes , we could for sure, I just thought it's cleaner and might avoid name clashes, but if it's a lot of works, then there is no need for the moment

Reply:
Potential name collisions have now been handled